### PR TITLE
Add initial feature tracking

### PR DESCRIPTION
**Pull request by andresmgot**
Query features from: #485

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -1,14 +1,59 @@
...
   .setConfigEditor(ConfigEditor)
   .setQueryEditorHelp(EditorHelp)
   .setQueryEditor(QueryEditor);
+
+// Track dashboard loads to RudderStack
+getAppEvents().subscribe<DashboardLoadedEvent<KustoQuery>>(
+  DashboardLoadedEvent,
+  ({ payload: { dashboardId, orgId, grafanaVersion, queries } }) => {
+    const adxQueries = queries[pluginJson.id]?.filter((q) => !q.hide);
+    if (!adxQueries?.length) {
```
Comment (aangelisc) [inline src/module.ts:24]:
```suggestion
    if (adxQueries && adxQueries.length > 0) {
```

As the condition around `trackADXMonitorDashboardLoaded` can be removed

File: src/tracking.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * Dashboard: https://ops.grafana.net/d/Ad0pti0N/data-sources-adoption-tracking?orgId=1
+ * Changelog:
+ * - 4.1.7: Initial version
+ */
+export const trackADXMonitorDashboardLoaded = (props: ADXMonitorDashboardLoadedProps) => {
+  reportInteraction('grafana_ds_adx_dashboard_loaded', props);
+};
+
+export type ADXMonitorDashboardLoadedProps = {
+  adx_plugin_version?: string;
```
Comment (yaelleC) [inline src/tracking.ts:21]: 🚀 nice!

File: src/module.ts [hunk 3]
```diff
@@ -1,14 +1,57 @@
...
 )
   .setConfigEditor(ConfigEditor)
   .setQueryEditorHelp(EditorHelp)
   .setQueryEditor(QueryEditor);
+
+// Track dashboard loads to RudderStack
+getAppEvents().subscribe<DashboardLoadedEvent<KustoQuery>>(
+  DashboardLoadedEvent,
+  ({ payload: { dashboardId, orgId, grafanaVersion, queries } }) => {
+    const adxQueries = queries[pluginJson.id]?.filter((q) => !q.hide);
```
Comment (yaelleC) [inline src/module.ts:23]: Keeping only the visible queries: sounds good!

**Review by yaelleC — APPROVED** [review]
LGTM
Just curious about the difference between `FormatOptions.adxTimeSeries` and `FormatOptions.TimeSeries` I thought we only had two possible format options ?

**Comment by andresmgot** [comment]
> Just curious about the difference between `FormatOptions.adxTimeSeries` and `FormatOptions.TimeSeries` I thought we only had two possible format options ?

It's kind of an advanced feature. That format is only available when using the KQL editor, since it requires to use the `make-series` Kusto operator, which is not something that you can add using the query builder. It's documented in our [docs](https://github.com/grafana/azure-data-explorer-datasource#adx-time-series-queries).

[Screencast from 10-10-22 14:48:22.webm](https://user-images.githubusercontent.com/4025665/194870079-cb011762-39a1-489e-a280-184b2671561c.webm)

**Pull request merged**